test/features/login/data/repository/credential_repository_imp_test.dart

```diff
@@ -0,0 +1,63 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/annotations.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+import 'package:tmail_ui_user/features/login/data/local/authentication_info_cache_manager.dart';
+import 'package:tmail_ui_user/features/login/data/repository/credential_repository_impl.dart';
+
+import 'credential_repository_imp_test.mocks.dart';
+
+@GenerateMocks([AuthenticationInfoCacheManager])
+void main() {
+  late SharedPreferences sharedPreferences;
+  late AuthenticationInfoCacheManager authenticationInfoCacheManager;
+  late CredentialRepositoryImpl credentialRepositoryImpl;
+
+  setUp(() async {
+    SharedPreferences.setMockInitialValues({});
+    sharedPreferences = await SharedPreferences.getInstance();
```

Comment:
IMO, It has to be initialized before `setMockInitialValues ` right?

Reply:
If we call it after getInstance. Error:
<img width="800" alt="image" src="https://github.com/user-attachments/assets/ba616030-f122-4c64-a659-e81647077269">

`SharedPreferences` is platform based class, so we need to set `initMock` before create this instance